spec/features/ingredients/index_spec.rb

```diff
@@ -0,0 +1,28 @@
+require "rails_helper"
+
+RSpec.describe "the ingredients index" do 
+  item_1 = Ingredient.create!(name: "salt", cost: 1)
+  item_2 = Ingredient.create!(name: "pepper", cost: 2)
+
+  # User Story 1 - Ingredients Index
+  it "lists all ingredients and their costs" do 
+    # When I visit '/ingredients'
+    visit "/ingredients"
+    # I see a list of all the ingredients including their name and cost
+    # (e.g. "Ground Beef: 2"
+    #      "Salt: 4")
+    expect(page).to have_content(item_1.name)
+    expect(page).to have_content(item_2.name)
+    expect(page).to have_content(item_1.cost)
+    expect(page).to have_content(item_2.cost)
```

Comment:
# Common IC Feedback Comments

### Adding unnecessary instance vars in a controller

i.e. in addition to setting up `@recipes`, the student also has `@ingredients = @recipe.ingredients`

While this isn’t wrong, it’s not necessary or conventional to create these additional instance variables as they are attributes for your first instance var. You can call `@recipe.ingredients` in the view and that is not a violation of MVC. What *would* violate MVC would be doing an AR query like `@recipe.find` or `@recipes.count`

### Non-MVC-compliant controller actions

This is not the most MVC-compliant controller organization. Keep in mind that we want to only use CRUD actions to name our controller actions. Consider instead . . .

### Using data logic in view

This is not the most MVC-compliant location for this logic. Even though it is a simple calculation, doing <`example query, like @recipes.count`> is an AR method that does actually manipulate a database query. Our model methods should be responsible for any AR/SQL logic beyond simple CRUD actions, so this logic should be encapsulated into a model method. 

### Robust Assertions with Labels

These assertions are good, but could be more robust if you anchored them to text they should appear next to. If you set up your view well with labels next to each of these attributes, you could change your assertions to     `expect(page).to have_content("Cost: #{ingrediet1.cost}")`. This would require some edits in your view, but they're very small

This will make sure that this data is located in the correct place, and is meaningfully rendered on the page. Also, if the cost were `7` and there happened to be a footer on the page that happened to include a 7, you could get a false positive by just asserting for the presence of `7`. So it's better to be specific about where the 7 appears and assert the presence of `Cost: 7`